hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -42,173 +38,241 @@ object HealthTransformations {
       )
     } else None
 
-  /** Parse all infectious disease related fields out of a raw RedCap record. */
-  def mapInfectiousDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
-    if (rawRecord.getBoolean("hs_dx_infectious_yn")) {
-      // iterate over all infectious diseases
-      infectious_diseases.flatMap {
-        // "other" case
-        case ("infect_other", categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            "dx_infect_other",
-            infectiousDiseaseCondition,
-            categorical,
-            conditionOtherDescription = rawRecord.getOptional("hs_dx_infect_other_spec")
-          )
-        // generic case for infectious disease
-        case (disease, categorical) =>
-          createHealthConditionRow(
-            rawRecord,
-            s"dx_${disease}",
-            infectiousDiseaseCondition,
-            categorical
-          )
-      }
-    } else None
-
-  def mapCongenitalEyeDisorder(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
-    if (rawRecord.getBoolean("hs_congenital_yn") && rawRecord.getBoolean("hs_cg_eye_disorders_yn"))
-      eye_cg_disorders.flatMap {
-        // "other" case
-        case ("other", categorical) =>
+  def processHealthCondition(
+    rawRecord: RawRecord,
+    healthCondition: HealthCondition
+  ): Iterable[HlesHealthCondition] = {
+    val congenital: Option[HlesHealthCondition] = if (healthCondition.congenital.isDefined) {
```

Comment:
Instead of `isDefined` here and `get` below, this is more idiomatic:
```scala
val congenital: Option[HlesHealthCondition] = healthCondition.congenital.flatMap {
  case (thing1, thing2) => ???

}
```

Reply:
THANK YOU, I knew this felt weird